Add initial set of suggested alerts

douglascamata opened the pull request:
* Alerts that might fire because of how tenants define their rules or Observatorium credentials are all set to `severity: "warning"`. 
* Alerts that might fire because of problems in `obsctl-reloader` itself are set to `severity: "critical"`. Examples: listing the `PrometheusRule` CRDs in the cluster and communicating with Observatorium's rules API.

Looking for a good review of the summary and descriptions, particularly for the string templating there.

coleenquadros commented on openshift/alerts.yaml:
```diff
@@ -0,0 +1,59 @@
+      > 0.10
+    for: 10m
+    labels:
+      severity: warning
```
What if multiple tenants are not able to authenticate, would we still keep it at warning?

douglascamata replied:
Yes. Unfortunately there's no good solution here that will be automatically good for every setup. It's even not possible to know in advance how many tenants there are in total to know whether it's 10%, 20%, or 100% of the tenants with issues (what would be the threshold to consider it more severe?). This is mostly an informative alert.

Plus authentication verification should live where it's verified, in Observatorium API.

thibaultmg reviewed (approved):
Looks good!

douglascamata commented:
I removed the alert `ObsCtlRulesStoreAuthenticationError`, that would always trigger when `ObsCtlRulesStoreServerError` triggers, as it matches on any 5xx or 4xx, generating some noise.

I also updated the ``ObsCtlRulesSetFailure` alert to not include the errors talking to the rules store API, which are also covered automatically by `ObsCtlRulesStoreServerError`. Now it only covers pre-processing that needs to be done before pushing the rules.

philipgough commented on openshift/alerts.yaml:
```diff
@@ -0,0 +1,45 @@
+  - alert: ObsCtlRulesSetFailure
+    annotations:
+      description: obsctl-reloader is failing to set rules for tenant {{ $labels.tenant }} before reaching Observatorium {{ $value | humanizePercentage }}% of the time due to {{ $labels.reason }}.
+      summary: Failing to set rules due to issue beforetalking to Observatorium.
```
```suggestion
before talking
```
